### PR TITLE
Normalize the shortened link message, and incorporate protocol prefix into the length check.

### PR DESCRIPTION
**Pull request by tangentstorm**

### CODE REVIEW

File: hamper/plugins/bitly.py [hunk 1]
```diff
@@ -84,12 +85,13 @@ def message(self, bot, comm):
...
+            template = "{nick}'s url: {url}"
             if title:
-                bot.reply(comm, "{url} - Title: {title}"
-                          .format(url=short_url, title=title))
-            else:
-                bot.reply(comm, "{nick}'s url: {url}"
-                          .format(nick=comm['user'], url=short_url))
+                template += " - Title: {title}"
+
+            bot.reply(comm, template).format(
```
Comment (chancez) [inline hamper/plugins/bitly.py]: Should be

```
bot.reply(comm, template.format(..))
# or
msg = template.format(..)
bot.reply(comm, msg)
```

**Comment by tangentstorm** [comment]
D'oh! I knew I wouldn't get away without actually running the thing. Fixing.

**Comment by tangentstorm** [comment]
okay, since I screwed up the last PR, I did it right this time. 
I refactored `bitly.py` to make it easier to test, and then wrote some unit tests.

File: hamper/plugins/bitly.py [hunk 2]
```diff
@@ -5,14 +5,72 @@
...
+        """Given a long URL, returns a bitly short URL"""
+        result = self._query('post', 'shorten', longUrl=long_url)
+        return result['data']['url']
+
+    def info(self, hash=None, short_url=None):
+        """
+        Given a hash or short_url, return all information bitly provides
+        """
+        params = {}
+        if not (hash or short_url):
```
Comment (chancez) [inline hamper/plugins/bitly.py]: I don't think you can use the word hash, as it is a python keyword.

**Comment by chancez** [comment]
Getting a traceback:
https://gist.github.com/ecnahc515/5239027

**Comment by tangentstorm** [comment]
I think this is all set.

I was going to move the exclude list into the configuration file but I didn't want to break tamper, so I just left is as a TODO.

Regarding `hash`: it's not really a keyword, just a normal function in `__builtins__`... There's so much stuff in there, it's kind of hard to avoid this issue sometimes... But: we're not using it and as far as I can tell, there's no reason we'd ever _need_ to use it, so I just removed it.

**Comment by chancez** [comment]
Good work.

**Pull request merged**